### ISSUE TITLE
No library header

### ISSUE THREAD

**Issue by bhawnamotwani** [ICRAR/libprofit#2]
Hello, when I make install libprofit, the library header file (library.h) never gets made. This causes an interruption when I try to setup pyprofit, since library.h is included in profit.h, which it tries to read.
I tried different C/C++ compilers to the same end. How can I resolve this?

<img width="547" alt="screen shot 1396-12-22 at 12 39 24 pm" src="https://user-images.githubusercontent.com/22461099/37356157-bd844926-26bb-11e8-834f-683be05df0c4.png">

<img width="926" alt="screen shot 1396-12-22 at 12 43 23 pm" src="https://user-images.githubusercontent.com/22461099/37356362-2b1a5cfa-26bc-11e8-98b3-bdf633604530.png">

**Comment by rtobar** [comment]
Hi Bhawna,

First of all, thanks for reporting this!

tl;dr: go to your libprofit repo, run `git checkout v1.6.0`, re-build and re-install `libprofit` and try again.

Now the longer explanation... by the looks of it, I assume you are using the latest git version of `libprofit`, right? In that case there was indeed a problem with the `library.h` header file not being installed. I've pushed a commit to fix that, thanks!

Having said that, you would still find problems building `pyprofit` against this particular version of the `libprofit`. `pyprofit` works with specific versions of `libprofit`, and after `1.6.0` was tagged in `libprofit` a big number of changes have been introduced in `libprofit` that `pyprofit` is not ready for yet. The fact that the latest code from the `libprofit` git repo still shows `1.6.0` as its version is actually misleading (it should be something like `1.7.0-dev` or something like that), and certainly leads `pyprofit` to think it can use it, while it actually cannot. I will definitely try to be more careful with this in the future.

As mentioned earlier, the solution is simple: build and instal *exactly* the `v1.6.0` version of `libprofit`. We ensure that works, but it will miss a few nicer items that have been added into what will become `1.7`.

**Comment by bhawnamotwani** [comment]
Thanks a lot for your prompt and helpful response!

**Comment by rtobar** [comment]
Glad that I could be of help.

Following what I mentioned earlier, I've improved now the versioning scheme of `libprofit` (now its version is `1.7.0-dev`), and also how `pyprofit` handles this extra bit of information. With the latest code in both repositories now `pyprofit` refuses to build against `libprofit` `1.7.0-dev`, giving a meaningful message about it.
